File: app/src/main/java/fr/corenting/edcompanion/models/CommodityFinderResult.java
package fr.corenting.edcompanion.models;

import org.threeten.bp.Instant;

public class CommodityFinderResult {

    public long BuyPrice;
    public String LandingPad;
    public String Station;
    public String System;
    public boolean PermitRequired;
    public boolean IsPlanetary;
    public long Stock;
    public int DistanceToStar;
    public float Distance;
    public Instant LastPriceUpdate;
    public int PriceDifferenceFromAverage;
}

File: app/src/main/java/fr/corenting/edcompanion/models/CommodityFinderResult.kt
package fr.corenting.edcompanion.models

import fr.corenting.edcompanion.models.apis.EDAPIV4.CommodityFinderResponse
import org.threeten.bp.DateTimeUtils
import org.threeten.bp.Instant

data class CommodityFinderResult(
    val price: Long,
    val landingPad: String,
    val station: String,
    val system: String,
    val isPlanetary: Boolean,
    val isSettlement: Boolean,
    val isFleetCarrier: Boolean,
    val quantity: Long,
    val distanceToArrival: Float,
    val distanceFromReferenceSystem: Float,
    val lastMarketUpdate: Instant,
    val pricePercentageDifference: Int
) {
    companion object {
        fun fromCommodityFinderResponse(res: CommodityFinderResponse): CommodityFinderResult {
            return CommodityFinderResult(
                res.Price,
                res.MaxLandingPad,
                res.Name,
                res.SystemName,
                res.IsPlanetary,
                res.IsSettlement,
                res.IsFleetCarrier,
                res.Quantity,
                res.DistanceToArrival,
                res.DistanceFromReferenceSystem,
                DateTimeUtils.toInstant(res.LastMarketUpdate),
                res.PricePercentageDifference
            )
        }
    }
}
